src/client/app/redux/api/metersApi.ts

```diff
@@ -0,0 +1,114 @@
+import { baseApi } from './baseApi';
+import { conversionsApi } from './conversionsApi';
+
+export const meterAdapter = createEntityAdapter<MeterData>({
+	sortComparer: (meterA, meterB) => meterA.identifier?.localeCompare(meterB.identifier, undefined, { sensitivity: 'accent' })
+
+})
+export const metersInitialState = meterAdapter.getInitialState()
+export type MeterDataState = EntityState<MeterData, number>
+
+export const metersApi = baseApi.injectEndpoints({
+	endpoints: builder => ({
+		getMeters: builder.query<MeterDataState, void>({
+			query: () => 'api/meters',
+			// Optional endpoint property that can transform incoming api responses if needed
+			// Use EntityAdapters from RTK to normalizeData, and generate commonSelectors
+			transformResponse: (response: MeterData[]) => {
+				return meterAdapter.setAll(metersInitialState, response.map(meter => ({
+					...meter,
+					readingFrequency: durationFormat(meter.readingFrequency)
+				})))
+			},
+			// Tags used for invalidation by mutation requests.
+			providesTags: ['MeterData']
+		}),
+		editMeter: builder.mutation<MeterData, { meterData: MeterData, shouldRefreshViews: boolean }>({
+			query: ({ meterData }) => ({
+				url: 'api/meters/edit',
+				method: 'POST',
+				body: { ...meterData }
+			}),
+			onQueryStarted: async ({ meterData, shouldRefreshViews }, { dispatch, queryFulfilled }) => {
+				queryFulfilled.then(() => {
+					// Update reading views if needed. Never redoCik so false.
+					dispatch(conversionsApi.endpoints.refresh.initiate({ redoCik: false, refreshReadingViews: shouldRefreshViews }))
+					dispatch(metersApi.util.updateQueryData('getMeters', undefined, cacheDraft => { meterAdapter.addOne(cacheDraft, meterData) }))
+
+				})
+			},
+			invalidatesTags: ['MeterData']
+		}),
+		addMeter: builder.mutation<MeterData, MeterData>({
+			query: meter => ({
+				url: 'api/meters/addMeter',
+				method: 'POST',
+				body: { ...meter }
+			}),
+			transformResponse: (data: MeterData) => ({ ...data, readingFrequency: durationFormat(data.readingFrequency) }),
+			onQueryStarted: (_arg, { dispatch, queryFulfilled }) => {
+				queryFulfilled.then(({ data }) => {
+					dispatch(metersApi.util.updateQueryData('getMeters', undefined, cacheDraft => { meterAdapter.addOne(cacheDraft, data) }))
+				})
+			}
+		}),
+		lineReadingsCount: builder.query<number, { meterIDs: number[], timeInterval: TimeInterval }>({
+			query: ({ meterIDs, timeInterval }) => `api/readings/line/count/meters/${meterIDs.join(',')}?timeInterval=${timeInterval.toString()}`
+		}),
+		details: builder.query<NamedIDItem[], void>({
+			query: () => 'api/meters'
+		}),
+		rawLineReadings: builder.query<RawReadings[], { meterID: number, timeInterval: TimeInterval }>({
+			query: ({ meterID, timeInterval }) => `api/readings/line/raw/meter/${meterID}?timeInterval=${timeInterval.toString()}`
+		})
+	})
+})
+
+
+export const selectMeterDataResult = metersApi.endpoints.getMeters.select()
+
```

Comment:
Is it standard practice to export even if pnly used in this file? Just trying to understand.

Reply:
Not sure if there's a standard practice here, or if there are performance implications. Just think its useful to have selectors accessible wherever if need.

Reply:
Maye see https://stackoverflow.com/questions/76705375/does-exporting-a-module-and-never-importing-it-anywhere-affect-performance

Reply:
Maybe see https://stackoverflow.com/questions/76705375/does-exporting-a-module-and-never-importing-it-anywhere-affect-performance. I'm not sure which way to go on this.

Reply:
I think this is a non-critical issue. I will deffer this to some later PR if desired.